### PR TITLE
Add IIS config for easier windows setup

### PR DESCRIPTION
**Pull request by tankerkiller125**
Add a `web.config` file to the public directory would do the same thing for IIS that .htaccess does for Apache in regards to automatically setting up the rewrite rules. I believe that this would drastically reduce the number of users having a hard time with IIS.

(HUGE thanks to https://discuss.flarum.org/d/24573-newbie-installation-guide-windows-w-plesk for the initial work on this, all credit goes to AntVincent, @katosdev and phenomlabs on the forums)

### CODE REVIEW

**Comment by luceos** [comment]
We declined a previous PR to add docker related files in the past. I think this PR falls in the same classification. It solves an issue only few people have, a solution that can be hosted by the community and something that brings additional effort on radical changes.

**Comment by clarkwinkelmann** [comment]
Is the sensitive file matching done with a regular expression? If so could we re-use the same apache/nginx regex so that if we update in the future, we can also easily change it here?

**Comment by tankerkiller125** [comment]
@clarkwinkelmann I do think that it is regex, so I think we could probably replace what's there with what's in apache/nginx

@luceos I think the difference with IIS config vs Docker is that docker is an entire hosting infrastructure that could possibly change often. the IIS config on the other hand is similar to the apache and nginx configs we already include in that they would rarely need to change.

**Comment by clarkwinkelmann** [comment]
It's probably also worth pointing out we comment out the sensitive rules by default for apache/nginx as you don't need them if you follow the public folder approach. So we probably also want them commented out here for consistency.

**Comment by tankerkiller125** [comment]
@clarkwinkelmann done and tested working on my work computer.

File: public/web.config [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+  <system.webServer>
+        <defaultDocument>
+            <files>
+                <clear />
+                <add value="Index.php" />
```
Comment (clarkwinkelmann) [inline public/web.config:7]: is it normal to have an uppercase "I" here?

Comment (tankerkiller125) [reply public/web.config:7]: Setting the default doc through the IIS GUI sets it to a capital "I" it's a case insensitive value.

Comment (clarkwinkelmann) [reply public/web.config:7]: Ok then Windows, really confusing to look at though :man_shrugging:

File: public/web.config [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+  <system.webServer>
```
Comment (franzliedke) [inline public/web.config:3]: That looks like it's not indented correctly.

And that's about as much review I can do for this feature. :laughing:

Comment (askvortsov1) [reply public/web.config:3]: Linux gang unite!

Comment (tankerkiller125) [reply public/web.config:3]: I'll re run through XML parser just to double check. Probably won't get to it tonight though.

**Comment by tankerkiller125** [comment]
I've made what will hopefully be the last commit. For one I ran it through a linter to make sure everything is indented nicely. Second instead of commenting out the no-public restrictions rule I've used a `enabled="false"` flag. This makes it much easier to enable the rule even from the IIS GUI, but by default keeps it turned off.

**Review by fadedmax — APPROVED** [review]
lgtm

**Pull request merged**